packages/message-manager/src/AbstractMessageManager.ts

```diff
@@ -105,7 +105,12 @@ export abstract class AbstractMessageManager<
     message.status = status;
     this.updateMessage(message);
     this.hub.emit(`${messageId}:${status}`, message);
-    if (status === 'rejected' || status === 'signed' || status === 'errored') {
+    if (
+      status === 'rejected' ||
+      status === 'signed' ||
+      status === 'errored' ||
+      status === 'received'
```

Comment:
Rather than couple this abstract to more domain, perhaps we add a new optional constructor argument like `additionalFinishStatuses` so we can include this dynamically?

Reply:
Done.